app/models/katello/upstream_pool.rb

```diff
@@ -38,20 +38,25 @@ def upstream_consumer_id
         CP_POOL.upstream_consumer_id
       end
 
-      def upstream_pool_id_map(local_pool_ids)
+      def upstream_pool_id_map(local_pool_ids, fail_on_not_found: true)
         return {} unless local_pool_ids
 
-        Katello::Candlepin::PoolService.local_to_upstream_ids(local_pool_ids)
+        Katello::Candlepin::PoolService.local_to_upstream_ids(local_pool_ids, fail_on_not_found: fail_on_not_found)
       end
 
       def response_to_pools(response, pool_id_map: {})
         pools = JSON.parse(response)
+        if pool_id_map.empty?
```

Comment:
This block of code looks like it belongs in another method in the PoolService class which would take the upstream_pool_ids as input and returns a hash structure like PoolService.local_to_upstream_ids.

If you want to go that far, cool! I think it will make changing this code in the future a lot easier since `response_to_pools` will have fewer responsibilities.

If you want to do something more middle-ground: consider changing the signature of PoolService.local_to_upstream_ids to accept local pools (array of Katello::Pool objects) *or* local pool ids to avoid the extra DB lookup you're doing here